config/fi_provider.m4

```diff
@@ -192,3 +193,56 @@ AC_DEFUN([_FI_PROVIDER_INVOKE],[
 		[])]
 	    )
 ])
+
+dnl
+dnl Internal; should not be called from provder .m4 scripts.
+dnl Helper macro that appends each providers build flags for pkg-config
+dnl
+dnl Arguments:
+dnl
+dnl $1: name of the provider
+dnl
+AC_DEFUN([_FI_PROVIDER_PC_ADD],[
+	 FI_PC_CFLAGS="$FI_PC_CFLAGS $$1_CFLAGS $$1_CPPFLAGS"
+	 FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $$1_LIBS"
+	 # usnic with nl needs special handling (should be added to usnic_LIBS?)
+	 AS_IF([test "$1" == "usnic"], [
+	  FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $$1_nl_LIBS"])
+])
+
+dnl
+dnl Substitute pkg-config variables after all providers have been initialized
+dnl If any package dependencies are ever needed, make use of
+dnl   FI_PC_REQUIRES_PKGS
+dnl   and
+dnl   FI_PC_PRIVATE_PKGS
+dnl
+AC_DEFUN([FI_PROVIDER_SETUP_PC],[
+	FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $LIBS"
+
+	dnl Try and avoid duplication of private library flags
+	dnl Inefficient but easier than tracking within each provider macro
+	seen=""
+	for arg in $FI_PC_PRIVATE_LIBS
+	do
+	  found=no
+	  for s in $seen
+	  do
+	    AS_IF([test "$s" = "$arg"], [found=yes])
+          done
+	  AS_IF([test "$found" = "no"], [seen="$seen $arg"])
+	done
+	FI_PC_PRIVATE_LIBS=$seen
+
+	AS_IF([test "x$enable_shared" != xyes],
```

Comment:
Should this test instead check if `$enable_static` is `yes`?

I ask because it's possible/valid to `--enable-static --enable-shared` (i.e., build *both* the static and shared libraries).

That being said, I'm not 100% sure what the Right thing is to do here -- if both the static and shared libraries are available/installed, the linker tends to prefer the shared libraries (unless told otherwise); does `pkg-config` allow you to select between static and shared linking?